src/views/CollectionIndex/useLocalStorage.ts

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react';
+
+function useLocalStorage(key, initialValue) {
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      const item = window.localStorage.getItem(key);
+      return item ? JSON.parse(item) : initialValue;
+    } catch (e) {
+      console.log(e);
+      return initialValue;
+    }
+  });
+  const setValue = value => {
+    try {
+      const valueToStore = value instanceof Function ? value(storedValue) : value;
+      setStoredValue(valueToStore);
+      window.localStorage.setItem(key, JSON.stringify(valueToStore));
+    } catch (e) {
+      console.log(e);
+    }
+  };
+  return [storedValue, setValue];
+}
+export default useLocalStorage;
```

Comment:
Actually we already have this in `@next/hooks/useLocalStorage`. Please look first next time so we don't repeat code.

Reply:
Sure Adrien